### ISSUE TITLE
Not working with new firmware Evolution/44.0

### ISSUE THREAD

**Issue by rankjie** [lkempf/casambi-bt#28]
First of all, thank you for your hard work on this project! It has hugely improved my life. Being able to control the lights with HomeKit and other integrations has been fantastic, and not having this ability now is quite a pain.

## Issue Description

I upgraded all lights' firmware from Evolution/43.0 to Evolution/44.0 today. After rebooting Home Assistant, the Casambi integration stopped working.

It was still working BEFORE rebooting, I could still control the lights. So I believe the error raises during init.

The integration now shows the following error:
```
Unexpected error creating network 2A:CC:3F:C4:84:12
```

## Debug Log

I enabled debug logging, and here's the relevant output:

```
Logger: homeassistant.config_entries
Source: config_entries.py:604
First occurred: 4:42:47 PM (2 occurrences)
Last logged: 5:29:19 PM
Error setting up entry *my network name* for casambi_bt: Unexpected error creating network 2A:CC:3F:C4:84:12
Traceback (most recent call last):
  File "/config/custom_components/casambi_bt/__init__.py", line 102, in connect
    await self.casa.connect(device, self.password)
  File "/usr/local/lib/python3.12/site-packages/CasambiBt/_casambi.py", line 147, in connect
    await self._connectClient()
  File "/usr/local/lib/python3.12/site-packages/CasambiBt/_casambi.py", line 158, in _connectClient
    raise e
  File "/usr/local/lib/python3.12/site-packages/CasambiBt/_casambi.py", line 154, in _connectClient
    await self._casaClient.exchangeKey(self._casaNetwork.getKeyStore())  # type: ignore[union-attr]
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/CasambiBt/_client.py", line 158, in exchangeKey
    raise ProtocolError(
CasambiBt.errors.ProtocolError: Unexpected answer from device! Wrong device or protocol version?
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 604, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/casambi_bt/__init__.py", line 31, in async_setup_entry
    await api.connect()
  File "/config/custom_components/casambi_bt/__init__.py", line 115, in connect
    raise ConfigEntryError(
homeassistant.exceptions.ConfigEntryError: Unexpected error creating network 2A:CC:3F:C4:84:12
```

## Additional Information

I've attempted to modify the code locally, but I'm new to the Home Assistant ecosystem and wasn't able to resolve the issue. I decided to post this issue for assistance.

Thank you for your help!

**Comment by lkempf** [comment]
I'm glad to hear that you like this library this much. Regarding your error it seems like Casambi modified the protocol (or more precisely increased the protocol version). This means I will have to find time to discover the changes and implement them. In the meantime I would recommend going back to the previous version.

**Comment by rankjie** [comment]
Thanks for the swift response.
I also thought downgrading to the old firmware would be the easiest thing to do. However, the app does not seem to offer an option to do so. Perhaps I'll have to contact Casambi support then.

**Comment by lkempf** [comment]
I thought it was possible with an older version of the app but I'm not sure and can't check right now.

**Comment by schoenis** [comment]
> @schoenis Your running into [lkempf/casambi-bt#28](https://github.com/lkempf/casambi-bt/issues/28). Using the new Casambi firmware won't work with the integration currently. Maybe that also has something to do with the fact that your network no longer uses email and password. Is it possible to change the authentication mode in the settings somewhere?

Unfortunately not...

**Comment by rankjie** [comment]
> > Thanks for the swift response. I also thought downgrading to the old firmware would be the easiest thing to do. However, the app does not seem to offer an option to do so. Perhaps I'll have to contact Casambi support then.
> 
> please tell me if there is a way to downgrade to the last firmware version!

I've submitted a ticket on Casambi's support website. Last time I emailed them about unpairing devices from previous owner, they helped me out with a super admin account. I'm betting that they are willing to help me this time too. I'll let you know if this worked out for me.

**Comment by lkempf** [comment]
Ok thanks to @rankjie I now know that the new version doesn't seem to change much except for the protocol version. I'll try to publish an updated version check in the next couple of days.

**Comment by rankjie** [comment]
That's brilliant news! I'm glad to have provided at least some useful information.
Thanks for your time maintaining the integration! @lkempf

**Comment by lkempf** [comment]
It should work as of 23570917e19d0797225810d5fb4d207937be8898. Can someone test that?

**Comment by rankjie** [comment]
Thank you for your response, Lukas. I've tested the changes, but I'm still encountering issues. Here's a summary of my testing process and the errors I encountered:
Testing steps:

1. SSH'd into HA VM, then ran docker exec -it homeassistant /bin/bash

2. In the shell, executed pip install -e git+https://github.com/lkempf/casambi-bt.git@2357091 (new dependency installed successfully)

3. Returned to HASS web interface and rebooted HASS

4. After reboot, the settings page showed casambi_bt was not loaded
<img width="331" alt="image" src="https://github.com/user-attachments/assets/a5fa105e-10db-43d2-93da-2bb606047b6f">

I checked the system log, here are the relavent errors:

<img width="721" alt="image" src="https://github.com/user-attachments/assets/5d26dfb8-0da3-458e-a086-4542f86e0aba">

Detailed Log of the `Detected blocking call to import_module...`
```

Logger: homeassistant.util.loop
Source: util/loop.py:77
First occurred: 10:23:20 PM (2 occurrences)
Last logged: 10:23:29 PM

Detected blocking call to import_module with args ('custom_components.casambi_bt',) in /usr/src/homeassistant/homeassistant/loader.py, line 1067: ComponentProtocol, importlib.import_module(self.pkg_path) inside the event loop; This is causing stability issues. Please create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue For developers, please see https://developers.home-assistant.io/docs/asyncio_blocking_operations/#import_module Traceback (most recent call last): File "<frozen runpy>", line 198, in _run_module_as_main File "<frozen runpy>", line 88, in _run_code File "/usr/src/homeassistant/homeassistant/__main__.py", line 223, in <module> sys.exit(main()) File "/usr/src/homeassistant/homeassistant/__main__.py", line 209, in main exit_code = runner.run(runtime_conf) File "/usr/src/homeassistant/homeassistant/runner.py", line 189, in run return loop.run_until_complete(setup_and_run_hass(runtime_config)) File "/usr/local/lib/python3.12/asyncio/base_events.py", line 674, in run_until_complete self.run_forever() File "/usr/local/lib/python3.12/asyncio/base_events.py", line 641, in run_forever self._run_once() File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1990, in _run_once handle._run() File "/usr/local/lib/python3.12/asyncio/events.py", line 88, in _run self._context.run(self._callback, *self._args) File "/usr/src/homeassistant/homeassistant/setup.py", line 165, in async_setup_component result = await _async_setup_component(hass, domain, config) File "/usr/src/homeassistant/homeassistant/setup.py", line 334, in _async_setup_component component = await integration.async_get_component() File "/usr/src/homeassistant/homeassistant/loader.py", line 1019, in async_get_component comp = self._get_component() File "/usr/src/homeassistant/homeassistant/loader.py", line 1067, in _get_component ComponentProtocol, importlib.import_module(self.pkg_path)
Detected blocking call to import_module with args ('custom_components.casambi_bt.config_flow',) in /usr/src/homeassistant/homeassistant/loader.py, line 1296: return importlib.import_module(f"{self.pkg_path}.{platform_name}") inside the event loop; This is causing stability issues. Please create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue For developers, please see https://developers.home-assistant.io/docs/asyncio_blocking_operations/#import_module Traceback (most recent call last): File "<frozen runpy>", line 198, in _run_module_as_main File "<frozen runpy>", line 88, in _run_code File "/usr/src/homeassistant/homeassistant/__main__.py", line 223, in <module> sys.exit(main()) File "/usr/src/homeassistant/homeassistant/__main__.py", line 209, in main exit_code = runner.run(runtime_conf) File "/usr/src/homeassistant/homeassistant/runner.py", line 189, in run return loop.run_until_complete(setup_and_run_hass(runtime_config)) File "/usr/local/lib/python3.12/asyncio/base_events.py", line 674, in run_until_complete self.run_forever() File "/usr/local/lib/python3.12/asyncio/base_events.py", line 641, in run_forever self._run_once() File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1990, in _run_once handle._run() File "/usr/local/lib/python3.12/asyncio/events.py", line 88, in _run self._context.run(self._callback, *self._args) File "/usr/src/homeassistant/homeassistant/util/async_.py", line 112, in sem_task return await task File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1296, in async_init flow, result = await self._async_init(flow_id, handler, context, data) File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1321, in _async_init flow = await self.async_create_flow(handler, context=context, data=data) File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1462, in async_create_flow handler = await _async_get_flow_handler( File "/usr/src/homeassistant/homeassistant/config_entries.py", line 2913, in _async_get_flow_handler await _load_integration(hass, domain, hass_config) File "/usr/src/homeassistant/homeassistant/config_entries.py", line 2892, in _load_integration await integration.async_get_platform("config_flow") File "/usr/src/homeassistant/homeassistant/loader.py", line 1100, in async_get_platform platforms = await self.async_get_platforms((platform_name,)) File "/usr/src/homeassistant/homeassistant/loader.py", line 1165, in async_get_platforms platforms.update(self._load_platforms(platform_names)) File "/usr/src/homeassistant/homeassistant/loader.py", line 1090, in _load_platforms platform_name: self._load_platform(platform_name) File "/usr/src/homeassistant/homeassistant/loader.py", line 1264, in _load_platform cache[full_name] = self._import_platform(platform_name) File "/usr/src/homeassistant/homeassistant/loader.py", line 1296, in _import_platform return importlib.import_module(f"{self.pkg_path}.{platform_name}")
```


Detailed log of `Error occurred loading flow for integration casambi_bt`
```
Logger: homeassistant.config_entries
Source: config_entries.py:2894
First occurred: 10:23:29 PM (1 occurrences)
Last logged: 10:23:29 PM

Error occurred loading flow for integration casambi_bt: cannot import name 'ConnectionState' from partially initialized module 'CasambiBt._client' (most likely due to a circular import) (/usr/local/lib/python3.12/src/casambi-bt/src/CasambiBt/_client.py)
```

Detailed log of `Error doing job: Task exception was never retrieved (None)`
```
Logger: homeassistant
Source: config_entries.py:2899
First occurred: 10:23:29 PM (1 occurrences)
Last logged: 10:23:29 PM

Error doing job: Task exception was never retrieved (None)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 2892, in _load_integration
    await integration.async_get_platform("config_flow")
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1100, in async_get_platform
    platforms = await self.async_get_platforms((platform_name,))
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1177, in async_get_platforms
    import_future.result()
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1165, in async_get_platforms
    platforms.update(self._load_platforms(platform_names))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1090, in _load_platforms
    platform_name: self._load_platform(platform_name)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1264, in _load_platform
    cache[full_name] = self._import_platform(platform_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1296, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 200, in protected_loop_func
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1310, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/casambi_bt/__init__.py", line 10, in <module>
    from CasambiBt import Casambi, Group, Scene, Unit, UnitControlType
  File "/usr/local/lib/python3.12/src/casambi-bt/src/CasambiBt/__init__.py", line 6, in <module>
    from ._casambi import Casambi
  File "/usr/local/lib/python3.12/src/casambi-bt/src/CasambiBt/_casambi.py", line 12, in <module>
    from ._client import CasambiClient, ConnectionState, IncommingPacketType
  File "/usr/local/lib/python3.12/src/casambi-bt/src/CasambiBt/_client.py", line 24, in <module>
    from ._network import Network
  File "/usr/local/lib/python3.12/src/casambi-bt/src/CasambiBt/_network.py", line 15, in <module>
    from .errors import (
  File "/usr/local/lib/python3.12/src/casambi-bt/src/CasambiBt/errors.py", line 5, in <module>
    from ._client import ConnectionState
ImportError: cannot import name 'ConnectionState' from partially initialized module 'CasambiBt._client' (most likely due to a circular import) (/usr/local/lib/python3.12/src/casambi-bt/src/CasambiBt/_client.py)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/discovery_flow.py", line 108, in _async_start
    await gather_with_limited_concurrency(FLOW_INIT_LIMIT, *init_coros)
  File "/usr/src/homeassistant/homeassistant/util/async_.py", line 114, in gather_with_limited_concurrency
    return await gather(
           ^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/async_.py", line 112, in sem_task
    return await task
           ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1296, in async_init
    flow, result = await self._async_init(flow_id, handler, context, data)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1321, in _async_init
    flow = await self.async_create_flow(handler, context=context, data=data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1462, in async_create_flow
    handler = await _async_get_flow_handler(
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 2913, in _async_get_flow_handler
    await _load_integration(hass, domain, hass_config)
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 2899, in _load_integration
    raise data_entry_flow.UnknownHandler from err
homeassistant.data_entry_flow.UnknownHandler
```

**Comment by lkempf** [comment]
Thas was stupid, sorry. Should be fixed now.

**Comment by rankjie** [comment]
Now it loads but new error:

```
Logger: homeassistant.config_entries
Source: config_entries.py:604
First occurred: 11:58:31 PM (2 occurrences)
Last logged: 11:59:37 PM

Error setting up entry hou for casambi_bt: Unexpected error creating network 2A:CC:3F:C4:84:12
Traceback (most recent call last):
  File "/config/custom_components/casambi_bt/__init__.py", line 102, in connect
    await self.casa.connect(device, self.password)
  File "/usr/local/lib/python3.12/src/casambi-bt/src/src/casambi-bt/src/CasambiBt/_casambi.py", line 144, in connect
    self._casaClient = CasambiClient(
                       ^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/src/casambi-bt/src/src/casambi-bt/src/CasambiBt/_client.py", line 54, in __init__
    self._checkProtocolVersion(network.protocolVersion)
  File "/usr/local/lib/python3.12/src/casambi-bt/src/src/casambi-bt/src/CasambiBt/_client.py", line 92, in _checkProtocolVersion
    self._logger.warning(
    ^^^^^^^^^^^^
AttributeError: 'CasambiClient' object has no attribute '_logger'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 604, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/casambi_bt/__init__.py", line 31, in async_setup_entry
    await api.connect()
  File "/config/custom_components/casambi_bt/__init__.py", line 115, in connect
    raise ConfigEntryError(
homeassistant.exceptions.ConfigEntryError: Unexpected error creating network 2A:CC:3F:C4:84:12
```

**Comment by rankjie** [comment]
Digged a bit into the code, and I see the const for MIN/MAX protocol version is the same, probably forgot to changed?
https://github.com/lkempf/casambi-bt/blob/0bc1c2a06ed59962c6981020ae476116d24c9dd5/src/CasambiBt/_client.py#L42

```
MIN_VERSION: Final[int] = 10
MAX_VERSION: Final[int] = 10
```

**Comment by lkempf** [comment]
And I've pushed another fix

**Comment by lkempf** [comment]
> Digged a bit into the code, and I see the const for MIN/MAX protocol version is the same, probably forgot to changed?
> 
> https://github.com/lkempf/casambi-bt/blob/0bc1c2a06ed59962c6981020ae476116d24c9dd5/src/CasambiBt/_client.py#L42
> 
> ```
> MIN_VERSION: Final[int] = 10
> MAX_VERSION: Final[int] = 10
> ```

No if MAX_VERSION is exceeded that just causes a warning

**Comment by rankjie** [comment]
Installed `d7b46f`, now it's not throwing erros but stucked at init.


Firstly saw expected warning:
```
Logger: CasambiBt._client
Source: src/casambi-bt/src/CasambiBt/_client.py:92
First occurred: 12:39:24 PM (1 occurrences)
Last logged: 12:39:24 PM

Version too new. Your network version is 11. Highest supported version is 10. Continue at your own risk.
```

Then an error:
```
Logger: CasambiBt._client
Source: src/casambi-bt/src/CasambiBt/_client.py:171
First occurred: 12:39:26 PM (1 occurrences)
Last logged: 12:39:26 PM

Unexpected answer from device! Wrong device or protocol version? Trying to continue.
```

```
Logger: CasambiBt._client
Source: src/casambi-bt/src/CasambiBt/_client.py:260
First occurred: 12:39:26 PM (1 occurrences)
Last logged: 12:39:26 PM

Unhandled notify in state 99: b'02fffda8a70733658885976b19547c7f953c584797cbe306898724c2b5aa6212c54253a9b162452843bc150da6d090300e53d87cc99737477d69263deb90689150'
```

It seems to become unresponsive from this point
```
Logger: homeassistant.setup
Source: runner.py:189
First occurred: 12:39:33 PM (1 occurrences)
Last logged: 12:39:33 PM

Setup of rest is taking over 10 seconds.
```

```
Logger: homeassistant.bootstrap
Source: bootstrap.py:738
First occurred: 12:40:20 PM (5 occurrences)
Last logged: 12:44:20 PM

Waiting on integrations to complete setup: {('casambi_bt', 'fc8461de92e186495147fdb327fddea9'): 158257.526058721}
```

Finally it finally gave up/killed by hass:
```
Logger: homeassistant.config_entries
Source: config_entries.py:604
First occurred: 12:44:21 PM (1 occurrences)
Last logged: 12:44:21 PM

Error setting up entry hou for casambi_bt
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 604, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/casambi_bt/__init__.py", line 31, in async_setup_entry
    await api.connect()
  File "/config/custom_components/casambi_bt/__init__.py", line 102, in connect
    await self.casa.connect(device, self.password)
  File "/config/src/casambi-bt/src/CasambiBt/_casambi.py", line 150, in connect
    await self._connectClient()
  File "/config/src/casambi-bt/src/CasambiBt/_casambi.py", line 157, in _connectClient
    await self._casaClient.exchangeKey()
  File "/config/src/casambi-bt/src/CasambiBt/_client.py", line 192, in exchangeKey
    await self._notifySignal.wait()
  File "/usr/local/lib/python3.12/asyncio/locks.py", line 212, in wait
    await fut
asyncio.exceptions.CancelledError: Global task timeout
```
```
Logger: homeassistant
Source: runner.py:147
First occurred: 12:44:22 PM (1 occurrences)
Last logged: 12:44:22 PM

Error doing job: Task was destroyed but it is pending! (<Task pending name='Task-264' coro=<CasambiClient._processCallbacks() running at /config/src/casambi-bt/src/CasambiBt/_client.py:236> wait_for=<Future pending cb=[Task.task_wakeup()]>>)
```

**Comment by lkempf** [comment]
And another fix. Sorry, I'm really bad at writing code when I can't test it myself.

**Comment by rankjie** [comment]
Haha, please don't be sorry. You're working on a fix that benefits a lot of other people, that's amazing work. Also, that's just how coding works, isn't it. lol

Good news, it's working all fine for me, incl. on/off, dimming. 

Btw, I noticed a warning in the log, I'll just leave it here since I don't want to post another issue in the -hass repo.
```
Logger: homeassistant.components.light
Source: components/light/__init__.py:1172
integration: Light (documentation, issues)
First occurred: 12:55:42 PM (30 occurrences)
Last logged: 12:55:42 PM

None (<class 'custom_components.casambi_bt.light.CasambiLightUnit'>) sets invalid supported color modes {'brightness', 'onoff'}, this will stop working in Home Assistant Core 2025.3, please report it to the author of the 'casambi_bt' custom integration
None (<class 'custom_components.casambi_bt.light.CasambiLightGroup'>) sets invalid supported color modes {'brightness', 'onoff'}, this will stop working in Home Assistant Core 2025.3, please report it to the author of the 'casambi_bt' custom integration
```

**Comment by schoenis** [comment]
Amateur question: How do I get the update installed? Thanks for your help and good work!

**Comment by rankjie** [comment]
@schoenis 
I'm not sure how you run HA. From what I've learnt, you can try:

1. SSH'd into HA VM or use the `SSH & Web Terminal` add-on (https://community.home-assistant.io/t/home-assistant-community-add-on-ssh-web-terminal/33820), after installing the add-on, head back to its settings(Settings - Add-ons - Advanced SSH & Web Terminal - disable Protection mode)

2 . In the terminal, run `docker exec -it homeassistant /bin/bash`

3. Then in the docker's shell, execute `pip install -e git+https://github.com/lkempf/casambi-bt.git#egg=casambi-bt`

4. Returned to HASS web interface, go to Developer tools - Restart

**Comment by Alexei-vk2** [comment]
the solution works

**Comment by lkempf** [comment]
> Amateur question: How do I get the update installed? Thanks for your help and good work!

The new release is out now. Just update with HACS.
